Stratis.Bitcoin/Features/BlockStore/LoopSteps/BlockStoreLoopStep.cs

```diff
@@ -0,0 +1,87 @@
+﻿using NBitcoin;
+using Stratis.Bitcoin.Utilities;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Stratis.Bitcoin.Features.BlockStore.LoopSteps
+{
+    internal sealed class BlockStoreLoopStepChain
+    {
+        private List<BlockStoreLoopStep> steps = new List<BlockStoreLoopStep>();
+        private bool disposeMode;
```

Comment:
I will create the disposeMode property as a argument to be passed into the Execute method()